### PR TITLE
feat: add User.updateDisplayName and User.updatePhotoURL

### LINKED ISSUE #6194: 🐛 [firebase_auth] updateProfile(photoURL: null) doesn't set photoURL to null on mobile
## Bug report

This simple code doesn't set photoURL to null in the firebase auth of the connected user:
```dart
await FirebaseAuth.instance.currentUser.updateProfile(photoURL: null);
```

Doing the same from admin SDK properly set photoURL to null:
```js
await admin.auth().updateUser(uid, {photoURL: null});
```

### Steps to reproduce

1. Connect a user to a flutter app using firebase auth (I use firebase_auth 1.2.0)
2. Make sure this user has a profile picture (photoURL not null)
3. Execute this code to check if photoURL is set to null
```dart
User _user = FirebaseAuth.instance.currentUser;
await _user.updateProfile(photoURL: null);
await _user.reload();

print(FirebaseAuth.instance.currentUser);
```
In my case, photoURL keeps its current value and is not set to null.

### Expected behavior

photoURL should be null

---

### Flutter doctor

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel stable, 2.0.6, on macOS 11.2.3 20D91 darwin-x64, locale fr)
    • Flutter version 2.0.6 at /Users/droduit/dev/flutter
    • Framework revision 1d9032c7e1 (3 weeks ago), 2021-04-29 17:37:58 -0700
    • Engine revision 05e680e202
    • Dart version 2.12.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/droduit/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.56.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.22.0

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 90.0.4430.212

• No issues found!
```

</details>

---

### Flutter dependencies

<details><summary>Click To Expand</summary>

```
Dart SDK 2.12.3
Flutter SDK 2.0.6

dependencies:
- app_settings 4.1.0 [flutter]
- booking 0.0.1 [flutter]
- carousel_slider 3.0.0 [flutter]
- chat 0.0.1 [flutter easy_localization firebase_auth firebase_core firebase_storage firebase_database cloud_firestore cloud_functions flutter_local_notifications file_picker rxdart linkable percent_indicator chewie just_audio native_pdf_renderer after_layout twilio_programmable_video recase native_device_orientation flutter_dotenv flutter_ringtone_player flutter_svg progress_indicators universal_html universal_ui common]
- cloud_firestore 2.2.0 [cloud_firestore_platform_interface cloud_firestore_web firebase_core firebase_core_platform_interface flutter meta]
- cloud_functions 1.1.0 [cloud_functions_platform_interface cloud_functions_web firebase_core firebase_core_platform_interface flutter]
- common 0.0.1 [flutter easy_localization firebase_auth firebase_core firebase_storage cloud_firestore universal_html google_fonts uuid mime permission_handler ext_storage device_info week_of_year shimmer]
- cupertino_icons 1.0.2
- dotted_border 2.0.0 [flutter path_drawing]
- easy_localization 3.0.0 [flutter shared_preferences intl args path easy_logger flutter_localizations]
- email_validator 2.0.1
- extended_image 4.1.0 [extended_image_library flutter meta]
- firebase_analytics 8.1.0 [firebase_analytics_platform_interface firebase_analytics_web firebase_core flutter meta]
- firebase_auth 1.2.0 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 1.2.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_crashlytics 2.0.4 [firebase_core firebase_core_platform_interface firebase_crashlytics_platform_interface flutter stack_trace]
- firebase_database 6.1.2 [firebase_core flutter]
- firebase_dynamic_links 2.0.4 [firebase_core flutter]
- firebase_messaging 10.0.0 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
- firebase_storage 8.1.0 [firebase_core firebase_core_platform_interface firebase_storage_platform_interface firebase_storage_web flutter]
- fl_chart 0.30.0 [flutter equatable pedantic]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- flutter_dotenv 3.1.0 [flutter meta]
- flutter_local_notifications 5.0.0+1 [flutter flutter_local_notifications_platform_interface platform timezone]
- flutter_secure_storage 4.2.0 [meta flutter]
- flutter_slidable 0.5.7 [flutter]
- flutter_svg 0.21.0+1 [flutter meta path_drawing vector_math xml]
- google_fonts 2.0.0 [flutter http path_provider crypto]
- google_sign_in 5.0.2 [google_sign_in_platform_interface google_sign_in_web flutter meta]
- http 0.13.1 [http_parser meta path pedantic]
- image_editor 1.0.0 [flutter]
- image_picker 0.7.4 [flutter flutter_plugin_android_lifecycle image_picker_platform_interface image_picker_for_web]
- intl 0.17.0 [clock path]
- jiffy 4.1.0 [intl]
- json_annotation 4.0.1
- local_auth 1.1.5 [flutter meta intl platform flutter_plugin_android_lifecycle]
- open_file 3.2.0 [flutter ffi]
- provider 5.0.0 [collection flutter nested]
- sentry_flutter 5.0.0 [flutter flutter_web_plugins sentry package_info_plus]
- shimmer 1.1.2 [flutter]
- universal_html 2.0.8 [async csslib charcode collection html meta source_span typed_data universal_io]
- url_launcher 6.0.3 [flutter url_launcher_platform_interface url_launcher_linux url_launcher_macos url_launcher_windows url_launcher_web]

dev dependencies:
- build_runner 1.12.2 [args async build build_config build_daemon build_resolvers build_runner_core code_builder collection crypto dart_style glob graphs http_multi_server io js logging meta mime path pedantic pool pub_semver pubspec_parse shelf shelf_web_socket stack_trace stream_transform timing watcher web_socket_channel yaml]
- flutter_launcher_icons 0.9.0 [args image path yaml]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]
- json_serializable 4.1.1 [analyzer build build_config collection json_annotation meta path source_gen]

dependency overrides:
- equatable 2.0.0 [collection meta]
- html2md 1.1.0 [html collection]
- path_drawing 0.5.0 [vector_math meta path_parsing flutter]

transitive dependencies:
- _fe_analyzer_shared 21.0.0 [meta]
- after_layout 1.1.0 [flutter]
- analyzer 1.5.0 [_fe_analyzer_shared cli_util collection convert crypto glob meta package_config path pub_semver source_span watcher yaml pedantic]
- archive 3.1.2 [crypto path]
- args 2.0.0
- async 2.5.0 [collection]
- audio_session 0.1.0 [flutter flutter_web_plugins rxdart]
- boolean_selector 2.1.0 [source_span string_scanner]
- build 2.0.1 [analyzer async convert crypto glob logging meta path]
- build_config 0.4.7 [checked_yaml json_annotation meta path pubspec_parse yaml]
- build_daemon 2.1.10 [built_collection built_value http_multi_server logging pedantic path pool shelf shelf_web_socket stream_transform watcher web_socket_channel]
- build_resolvers 2.0.0 [analyzer build crypto graphs logging path package_config pool pub_semver stream_transform]
- build_runner_core 6.1.12 [async build build_config build_resolvers collection convert crypto glob graphs logging meta path package_config pedantic pool timing watcher yaml]
- built_collection 5.0.0
- built_value 8.0.5 [built_collection collection fixnum]
- characters 1.1.0
- charcode 1.2.0
- checked_yaml 2.0.1 [json_annotation source_span yaml]
- chewie 1.0.0 [cupertino_icons flutter video_player wakelock]
- cli_util 0.3.0 [meta path]
- clock 1.1.0
- cloud_firestore_platform_interface 5.1.0 [collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 2.1.0 [cloud_firestore_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- cloud_functions_platform_interface 5.0.5 [firebase_core flutter meta plugin_platform_interface]
- cloud_functions_web 4.0.7 [cloud_functions_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- code_builder 3.7.0 [built_collection built_value collection matcher meta]
- collection 1.15.0
- convert 3.0.0 [typed_data]
- crypto 3.0.1 [collection typed_data]
- csslib 0.17.0 [source_span]
- dart_style 2.0.0 [analyzer args path pub_semver source_span]
- dartlin 0.5.2
- device_info 2.0.0 [flutter device_info_platform_interface]
- device_info_platform_interface 2.0.1 [flutter meta plugin_platform_interface]
- easy_logger 0.0.2 [flutter]
- enum_to_string 1.0.14
- ext_storage 1.0.3 [flutter]
- extended_image_library 3.1.0 [crypto flutter http_client_helper path path_provider]
- extension 0.2.0
- fake_async 1.2.0 [clock collection]
- ffi 1.0.0
- file 6.1.0 [meta path]
- file_picker 3.0.1 [flutter flutter_web_plugins flutter_plugin_android_lifecycle plugin_platform_interface]
- firebase 9.0.1 [http http_parser js]
- firebase_analytics_platform_interface 2.0.1 [flutter meta]
- firebase_analytics_web 0.3.0+1 [firebase firebase_analytics_platform_interface flutter flutter_web_plugins meta]
- firebase_auth_platform_interface 4.2.3 [firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 1.1.3 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser intl js meta]
- firebase_core_platform_interface 4.0.1 [collection flutter meta plugin_platform_interface]
- firebase_core_web 1.1.0 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_crashlytics_platform_interface 3.0.4 [collection firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_platform_interface 3.0.0 [firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_web 2.0.0 [firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins js meta]
- firebase_storage_platform_interface 2.1.0 [collection firebase_core flutter meta plugin_platform_interface]
- firebase_storage_web 1.1.0 [async firebase_core firebase_core_web firebase_storage_platform_interface flutter flutter_web_plugins http js meta]
- fixnum 1.0.0
- flutter_local_notifications_platform_interface 3.0.0 [flutter plugin_platform_interface]
- flutter_localizations 0.0.0 [flutter intl characters clock collection meta path typed_data vector_math]
- flutter_plugin_android_lifecycle 2.0.1 [flutter]
- flutter_ringtone_player 2.0.0 [flutter]
- flutter_web_plugins 0.0.0 [flutter js characters collection meta typed_data vector_math]
- glob 2.0.1 [async collection file path pedantic string_scanner]
- google_sign_in_platform_interface 2.0.1 [flutter meta quiver]
- google_sign_in_web 0.10.0 [google_sign_in_platform_interface flutter flutter_web_plugins meta js]
- graphs 1.0.0
- html 0.15.0 [csslib source_span]
- http_client_helper 2.0.2 [http]
- http_multi_server 3.0.1 [async]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- image 3.0.2 [archive meta xml]
- image_picker_for_web 2.0.0 [image_picker_platform_interface meta flutter flutter_web_plugins]
- image_picker_platform_interface 2.1.0 [flutter meta http plugin_platform_interface]
- io 1.0.0 [meta path string_scanner]
- js 0.6.3
- just_audio 0.7.4+1 [just_audio_platform_interface just_audio_web audio_session rxdart path path_provider async uuid crypto meta flutter]
- just_audio_platform_interface 3.0.0 [flutter plugin_platform_interface]
- just_audio_web 0.3.1 [just_audio_platform_interface flutter flutter_web_plugins]
- linkable 1.0.9 [flutter url_launcher]
- logging 1.0.1
- matcher 0.12.10 [stack_trace]
- meta 1.3.0
- mime 1.0.0
- native_device_orientation 1.0.0 [flutter meta]
- native_pdf_renderer 3.0.0 [flutter flutter_web_plugins device_info uuid image js meta extension synchronized]
- nested 1.0.0 [flutter]
- package_config 2.0.0 [path]
- package_info_plus 1.0.0 [flutter package_info_plus_platform_interface package_info_plus_linux package_info_plus_macos package_info_plus_windows package_info_plus_web]
- package_info_plus_linux 1.0.0 [package_info_plus_platform_interface flutter path]
- package_info_plus_macos 1.1.0 [flutter]
- package_info_plus_platform_interface 1.0.0 [flutter meta plugin_platform_interface]
- package_info_plus_web 1.0.0 [http package_info_plus_platform_interface flutter flutter_web_plugins meta]
- package_info_plus_windows 1.0.0 [package_info_plus_platform_interface ffi flutter win32]
- path 1.8.0
- path_parsing 0.2.0 [vector_math meta]
- path_provider 2.0.1 [flutter path_provider_platform_interface path_provider_macos path_provider_linux path_provider_windows]
- path_provider_linux 2.0.0 [path xdg_directories path_provider_platform_interface flutter]
- path_provider_macos 2.0.0 [flutter]
- path_provider_platform_interface 2.0.1 [flutter meta platform plugin_platform_interface]
- path_provider_windows 2.0.1 [path_provider_platform_interface meta path flutter ffi win32]
- pedantic 1.11.0
- percent_indicator 3.0.1 [flutter]
- permission_handler 6.1.3 [flutter meta permission_handler_platform_interface]
- permission_handler_platform_interface 3.3.0 [flutter meta plugin_platform_interface]
- petitparser 4.1.0 [meta]
- platform 3.0.0
- plugin_platform_interface 2.0.0 [meta]
- pool 1.5.0 [async stack_trace]
- process 4.2.1 [file path platform]
- progress_indicators 1.0.0 [flutter]
- pub_semver 2.0.0 [collection]
- pubspec_parse 1.0.0 [checked_yaml collection json_annotation pub_semver yaml]
- quiver 3.0.1 [matcher]
- recase 3.0.1
- rxdart 0.26.0
- sentry 5.0.0 [http meta stack_trace uuid]
- shared_preferences 2.0.5 [meta flutter shared_preferences_platform_interface shared_preferences_linux shared_preferences_macos shared_preferences_web shared_preferences_windows]
- shared_preferences_linux 2.0.0 [flutter file meta path path_provider_linux shared_preferences_platform_interface]
- shared_preferences_macos 2.0.0 [shared_preferences_platform_interface flutter]
- shared_preferences_platform_interface 2.0.0 [flutter]
- shared_preferences_web 2.0.0 [shared_preferences_platform_interface flutter flutter_web_plugins meta]
- shared_preferences_windows 2.0.0 [shared_preferences_platform_interface flutter file meta path path_provider_platform_interface path_provider_windows]
- shelf 1.1.0 [async collection http_parser path stack_trace stream_channel]
- shelf_web_socket 1.0.1 [shelf stream_channel web_socket_channel]
- sky_engine 0.0.99
- source_gen 1.0.0 [analyzer async build dart_style glob meta path pedantic source_span]
- source_span 1.8.0 [charcode collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- stream_transform 2.0.0
- string_scanner 1.1.0 [charcode source_span]
- synchronized 3.0.0
- term_glyph 1.2.0
- test_api 0.2.19 [async boolean_selector collection meta path source_span stack_trace stream_channel string_scanner term_glyph matcher]
- timezone 0.7.0 [path]
- timing 1.0.0 [json_annotation]
- twilio_programmable_video 0.7.1 [flutter permission_handler enum_to_string twilio_programmable_video_platform_interface]
- twilio_programmable_video_platform_interface 0.3.1 [flutter meta enum_to_string plugin_platform_interface collection dartlin equatable]
- typed_data 1.3.0 [collection]
- universal_io 2.0.4 [collection crypto meta typed_data]
- universal_ui 0.0.8 [flutter]
- url_launcher_linux 2.0.0 [flutter]
- url_launcher_macos 2.0.0 [flutter]
- url_launcher_platform_interface 2.0.2 [flutter plugin_platform_interface]
- url_launcher_web 2.0.0 [url_launcher_platform_interface meta flutter flutter_web_plugins]
- url_launcher_windows 2.0.0 [flutter]
- uuid 3.0.4 [crypto]
- vector_math 2.1.0
- video_player 2.1.1 [meta video_player_platform_interface video_player_web flutter flutter_test]
- video_player_platform_interface 4.1.0 [flutter meta flutter_test]
- video_player_web 2.0.0 [flutter flutter_web_plugins meta video_player_platform_interface]
- wakelock 0.4.0 [flutter meta wakelock_macos wakelock_platform_interface wakelock_web]
- wakelock_macos 0.1.0 [flutter flutter_web_plugins wakelock_platform_interface]
- wakelock_platform_interface 0.2.0 [flutter meta]
- wakelock_web 0.2.0 [flutter flutter_web_plugins js wakelock_platform_interface]
- watcher 1.0.0 [async path pedantic]
- web_socket_channel 2.0.0 [async crypto stream_channel]
- week_of_year 1.0.1
- win32 2.0.5 [ffi]
- xdg_directories 0.2.0 [meta path process]
- xml 5.1.0 [collection meta petitparser]
- yaml 3.1.0 [collection source_span string_scanner]
```

</details>

---

### LINKED ISSUE #6199: 🐛 [firebase_auth] Changing the User's photoURL/displayName on web changes sets the unchanged variable to null
### Steps to reproduce

On web only:

- set a displayName and photoURL for the current user:
  ```dart
  await FirebaseAuth.instance.currentUser.updateProfile(
    displayName: 'Puppy',
    photoURL: 'https://upload.wikimedia.org/wikipedia/commons/6/6e/Golde33443.jpg,
  );
  ```

- change once more either the displayName or photoURL **but not both**:
  ```dart
  await FirebaseAuth.instance.currentUser.updateProfile(
    displayName: 'Remi',
  );
  ```
- the parameter that was not passed (here `photoURL`) is now `null` instead of its previous value

This behavior is inconsistent with mobile, where on mobile the value is preserved.

### PR DESCRIPTION
**Pull request by rrousselGit**
## Description

User.updateProfile is now deprecated in favor of these new methods.
This change is to fix an issue where it was not possible to set a displayName/photoURL to null.

## Related Issues

fixes #6199
fixes #6194

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_auth/firebase_auth/ios/Classes/FLTFirebaseAuthPlugin.m [hunk 1]
```diff
@@ -839,13 +839,23 @@ - (void)userUpdateProfile:(id)arguments withMethodCallResult:(FLTFirebaseMethodC
...
+    } else {  
+      changeRequest.displayName = profileUpdates[@"displayName"];
+    }
   }
 
-  if (profileUpdates[@"photoURL"] != nil && ![profileUpdates[@"photoURL"] isEqual:[NSNull null]]) {
-    changeRequest.photoURL = [NSURL URLWithString:profileUpdates[@"photoURL"]];
+  if (profileUpdates[@"photoURL"] != nil) {
+    if ([profileUpdates[@"photoURL"] isEqual:[NSNull null]]) {
+      // We apparently cannot set photoURL to nil/NULL to remove it.
```
Comment (russellwheatley) [inline packages/firebase_auth/firebase_auth/ios/Classes/FLTFirebaseAuthPlugin.m:852]: There are undocumented methods for setting these properties:
```obj-c
[changeRequest setDisplayName:nil];
[changeRequest setPhotoURL:nil];
```
that should work. Reference [here](https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseAuth/Sources/User/FIRUser.m#L1521-L1548)

File: packages/firebase_auth/firebase_auth/example/test_driver/user_e2e.dart [hunk 2]
```diff
@@ -560,16 +560,148 @@ void runUserTests() {
...
+          () async {
+        // First create a user with a photo
+        await FirebaseAuth.instance.createUserWithEmailAndPassword(
+          email: email,
+          password: testPassword,
+        );
+        await FirebaseAuth.instance.currentUser.updateDisplayName('Mona Lisa');
+        await FirebaseAuth.instance.currentUser.updatePhotoURL(
+          'http://photo.url/test.jpg',
+          // await Future.wait([
```
Comment (russellwheatley) [inline packages/firebase_auth/firebase_auth/example/test_driver/user_e2e.dart:574]: remove dead code.

File: packages/firebase_auth/firebase_auth/ios/Classes/FLTFirebaseAuthPlugin.m (hunk 1 above)
Comment (rrousselGit) [reply packages/firebase_auth/firebase_auth/ios/Classes/FLTFirebaseAuthPlugin.m:852]: It doesn't seem to work either

**Review by russellwheatley — CHANGES_REQUESTED** [review]
a couple of bits of feedback.

File: packages/firebase_auth/firebase_auth/ios/Classes/FLTFirebaseAuthPlugin.m (hunk 1 above)
Comment (russellwheatley) [reply packages/firebase_auth/firebase_auth/ios/Classes/FLTFirebaseAuthPlugin.m:852]: fair enough, I'll resolve this then 👍

**Comment by rrousselGit** [comment]
Blocked by https://github.com/firebase/firebase-ios-sdk/issues/8149

**Comment by rrousselGit** [comment]
@Salakar @russellwheatley should we merge this and skip the displayName = null test, while tracking the iOS bug on a separate issue?

Because I'm not sure it makes sense to block this when this correctly fixes the issue for Android/Web and iOS's photoURL.
And we're already unable to set displayname to `null` on IOS, so nothing new.

**Comment by russellwheatley** [comment]
> @Salakar @russellwheatley should we merge this and skip the displayName = null test, while tracking the iOS bug on a separate issue?

> Because I'm not sure it makes sense to block this when this correctly fixes the issue for Android/Web and iOS's photoURL.
And we're already unable to set displayname to null on IOS, so nothing new.

@rrousselGit makes sense to me 👍

**Comment by Salakar** [comment]
> @Salakar @russellwheatley should we merge this and skip the displayName = null test, while tracking the iOS bug on a separate issue?
> 
> Because I'm not sure it makes sense to block this when this correctly fixes the issue for Android/Web and iOS's photoURL.
> And we're already unable to set displayname to `null` on IOS, so nothing new.

SGTM

File: packages/firebase_auth/firebase_auth/example/test_driver/user_e2e.dart [hunk 3]
```diff
@@ -560,16 +560,147 @@ void runUserTests() {
...
+
+        await FirebaseAuth.instance.currentUser.updateDisplayName(null);
+        await FirebaseAuth.instance.currentUser.reload();
+
+        expect(
+          FirebaseAuth.instance.currentUser.displayName,
+          isNull,
+        );
+        // blocked by https://github.com/firebase/firebase-ios-sdk/issues/8149
+      }, skip: true);
```
Comment (russellwheatley) [inline packages/firebase_auth/firebase_auth/example/test_driver/user_e2e.dart:622]: shouldn't this just skip for `iOS`?

**Review by russellwheatley — APPROVED** [review]
apart from perhaps only skipping a test for `iOS`, this LGTM.

**Review by russellwheatley — APPROVED** [review]
LGTM

**Pull request merged**